tools/checklocks/checklocks.go

```diff
@@ -82,6 +110,7 @@ func run(pass *analysis.Pass) (interface{}, error) {
 		exemptions: make(map[positionKey]struct{}),
 		forced:     make(map[positionKey]struct{}),
 		functions:  make(map[*ssa.Function]struct{}),
+		perfdata:   newPkgPerfData(),
```

Comment:
If you want to be really careful about overhead, I would recommend having this be optionally nil.

```go
pc :=  ...
if Benchmark {
  pc.perf = &PkgPerfData{ ... }
}
```